File: app/src/release/java/com/fastaccess/provider/fabric/FabricProvider.java
package com.fastaccess.provider.fabric;

import android.content.Context;
import android.support.annotation.NonNull;

import com.fastaccess.BuildConfig;

/**
 * Created by kosh on 14/08/2017.
 */

public class FabricProvider {

    public static void initFabric(@NonNull Context context) {
        Fabric fabric = new Fabric.Builder(context)
                .kits(new Crashlytics.Builder()
                        .core(new CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG).build())
                        .build())
                .debuggable(BuildConfig.DEBUG)
                .build();
        Fabric.with(fabric);
    }

    public static void logPurchase(@NonNull String productKey) {
        Answers.getInstance().logPurchase(PurchaseEvent().putItemName(productKey).putSuccess(true));
    }
}

File: app/src/main/java/com/fastaccess/github/platform/fabric/FabricProvider.kt
package com.fastaccess.github.platform.fabric

import android.content.Context
import com.crashlytics.android.BuildConfig
import com.crashlytics.android.Crashlytics
import com.crashlytics.android.answers.Answers
import com.crashlytics.android.answers.PurchaseEvent
import com.crashlytics.android.core.CrashlyticsCore
import io.fabric.sdk.android.Fabric
import java.math.BigDecimal
import java.math.RoundingMode

object FabricProvider {

    fun initFabric(context: Context) {
        val fabric = Fabric.Builder(context)
                .kits(Crashlytics.Builder()
                        .core(CrashlyticsCore.Builder().disabled(BuildConfig.DEBUG).build())
                        .build())
                .debuggable(BuildConfig.DEBUG)
                .build()
        Fabric.with(fabric)
    }

    fun logPurchase(productKey: String, price: Long? = null, priceText: String? = null) {
        val purchaseEvent = PurchaseEvent()
                .putItemName(productKey)
                .putSuccess(true)
        priceText?.let { purchaseEvent.putItemType(it) }
        price?.let { purchaseEvent.putItemPrice(BigDecimal(it).setScale(2, RoundingMode.CEILING)) }
        Answers.getInstance().logPurchase(purchaseEvent)
    }
}
